include/threepp/core/Object3D.hpp

```diff
@@ -254,7 +254,12 @@ namespace threepp {
 
         virtual std::shared_ptr<Object3D> clone(bool recursive = true);
 
-        ~Object3D() override;
+        EventDispatcher OnAdded;
+        EventDispatcher OnRemove;
+        EventDispatcher OnDispose;
```

Comment:
I would have preferred to keep this generic. I.e. `OnDispose` is not apart of the Object3D API.
How is other types (user defined of future additions) handled?